files-community/Files
# Code quality: Move Git operations to a different thread

**Resolved / Related Issues**
   Related #12737 

**Changes**
- Branches are enumerated and filtered on a different thread
- `Head branch` name is now "cached"
- `Fetch, Pull, Push, Sync, Checkout` are now executed on a different thread.

**Note**
You may need to log-in github from `Files Preview or Stable` to use some functionalities

**Validation**
How did you test these changes?
   1. Check that Files doesn't freeze when opening large git repos
   2. Check that git actions (fetch, pull, push, checkout, sync) still work
